pennylane/devices/qubit/simulate.py

```diff
@@ -46,6 +47,59 @@
 }
 
 
+def _valid_flex_int(s):
+    """Returns True if s is a non-negative integer."""
+    return isinstance(s, int) and s >= 0
+
+
+def _valid_flex_tuple(s):
+    """Returns True if s is a tuple of the form (shots, copies)."""
+    return (
+        isinstance(s, tuple)
+        and len(s) == 2
+        and _valid_flex_int(s[0])
+        and isinstance(s[1], int)
+        and s[1] > 0
+    )
+
+
+class _FlexShots(qml.measurements.Shots):
+    """Shots class that allows zero shots."""
+
+    total_shots = None
```

Comment:
Can we potentially just inherit most of this behavior? We only really need to override behavior in `__init__` for `shots=0`.

Reply:
We could even just have a `ZeroShots` class that inherits from `Shots` to only represent the `shots=0` case.

Reply:
We want to allow shot vectors such as (1, 4, 0, 2, 0). Not each item in the shot vector will necessarily be zero.